docs/CustomApp.md

```diff
@@ -3,110 +3,244 @@ layout: default
+
+```js
+// in src/App.js
+import React from 'react';
+import { Provider } from 'react-redux';
+import createHistory from 'history/createHashHistory';
+import { Admin, Resource } from 'react-admin';
 import restProvider from 'ra-data-simple-rest';
 import defaultMessages from 'ra-language-english';
 
+import createAdminStore from './createAdminStore';
+import messages from './i18n';
+
 // your app components
 import Dashboard from './Dashboard';
 import { PostList, PostCreate, PostEdit, PostShow } from './Post';
 import { CommentList, CommentEdit, CommentCreate } from './Comment';
 import { UserList, UserEdit, UserCreate } from './User';
-// your app labels
-import messages from './i18n';
 
-// create a Redux app
-const reducer = combineReducers({
-    admin: adminReducer,
-    i18n: i18nReducer('en', messages['en']),
-    form: formReducer,
-    routing: routerReducer,
-});
-const sagaMiddleware = createSagaMiddleware();
+// side effects
+const authProvider = () => Promise.resolve();
+const dataProvider = restProvider('http://path.to.my.api/');
+const i18nProvider = locale => {
+    if (locale !== 'en') {
+        return messages[locale];
+    }
+    return defaultMessages;
+};
 const history = createHistory();
-const store = createStore(reducer, undefined, compose(
-    applyMiddleware(sagaMiddleware, routerMiddleware(history)),
-    window.devToolsExtension ? window.devToolsExtension() : f => f,
-));
+
+const App = () => (
+    <Provider
+        store={createAdminStore({
+            authProvider,
+            dataProvider,
+            i18nProvider,
+            history,
+        })}
+    >
+        <Admin
+            authProvider={authProvider}
+            history={history}
+            title="My Admin"
+        >
+            <Resource name="posts" list={PostList} create={PostCreate} edit={PostEdit} show={PostShow} />
+            <Resource name="comments" list={CommentList} edit={CommentEdit} create={CommentCreate} />
+            <Resource name="users" list={UserList} edit={UserEdit} create={UserCreate} />
+        </Admin>
+    </Provider>
+);
+
+export default App;
+```
+
+**Tip**: One thing to pay attention to is that you must pass the same `history` and `authProvider` to both the redux Store creator and the `<Admin>` component. But you don't need to pass the `dataProvider` or the `i18nProvider`.
+
+## Not Using the `<Admin>` Components
+
+The `<Admin>` component takes care of defining the store (unless you provide one, as seen above), of setting the Translation and Authentication contexts, and of bootstrapping the Router. In case you need to override any of these, you can use your own component instead of `<Admin>`. 
+
+Here is the main code for bootstrapping a barebones react-admin application without `<Admin>`:
+
+```diff
+// in src/App.js
+import React from 'react';
+import { Provider } from 'react-redux';
+import createHistory from 'history/createHashHistory';
++import { ConnectedRouter } from 'react-router-redux';
```

Comment:
I realize a diff is hard to read in a diff. As this chapter was almost completely rewritten, you can read the last version directly.